### PR TITLE
[Docs] add advanced_guides/transforms.md

### PR DESCRIPTION
**Pull request by wangbo-zhao**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

add docs/en/advanced_guides/transforms.md

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by wangbo-zhao** [comment]
I will run the pre-commit hook to pass the lint when the review finished.

File: docs/en/advanced_guides/transforms.md [hunk 1]
```diff
@@ -1 +1,59 @@
...
+`transforms`. There are three main categories of `transforms` in MMSelfSup:
+1. Transforms about processing the data. The unique transforms in MMSelfSup are defined in [processing.py](../../../mmselfsup/datasets/transforms/processing.py), e.g. `RandomCrop`, `RandomResizedCrop` and `RandomGaussianBlur`.
+We may also use some transforms from other repositories, e.g. `LoadImageFromFile` from MMCV.
+2. The transform wrapper for multiple views of an image. It is defined in [wrappers.py](../../../mmselfsup/datasets/transforms/wrappers.py).
+3. The transform to pack data into a format compatible with the inputs of the algorithm. It is defined in [formatting.py](../../../mmselfsup/datasets/transforms/formatting.py).
+
+The last two transforms will be introduced below.
+
+## Introduction of `MultiView`
+We build a wrapper named `MultiView` for some algorithms e.g. MOCO, SimCLR and SwAV with multi-view image inputs. In the config file, we can
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/transforms.md]:
```suggestion
We build a wrapper named [`MultiView`](mmselfsup.datasets.transforms.MultiView) for some algorithms e.g. MOCO, SimCLR and SwAV with multi-view image inputs. In the config file, we can 
```

File: docs/en/advanced_guides/transforms.md [hunk 2]
```diff
@@ -1 +1,59 @@
...
+- [Transforms]()
+  - [Overview of transforms](#overview-of-transforms)
+  - [Introduction of `MultiView`](#introduction-of-multiview)
+  - [Introduction of `PackSelfSupInputs`](#introduction-of-packselfsupinputs)
+
+
+## Overview of transforms
+We have introduced how to build a `Pipeline` in [add_transforms](./add_transforms.md). A `Pipeline` contains a series of
+`transforms`. There are three main categories of `transforms` in MMSelfSup:
+1. Transforms about processing the data. The unique transforms in MMSelfSup are defined in [processing.py](../../../mmselfsup/datasets/transforms/processing.py), e.g. `RandomCrop`, `RandomResizedCrop` and `RandomGaussianBlur`.
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/transforms.md]:
```suggestion
1. Transforms about processing the data. The unique transforms in MMSelfSup are defined in [processing.py](https://github.com/open-mmlab/mmselfsup/blob/1.x/mmselfsup/datasets/transforms/processing.py), e.g. `RandomCrop`, `RandomResizedCrop` and `RandomGaussianBlur`.
```

Comment (fangyixiao18) [reply docs/en/advanced_guides/transforms.md]: use github link instead of relative path

File: docs/en/advanced_guides/transforms.md [hunk 3]
```diff
@@ -1 +1,59 @@
...
+We may also use some transforms from other repositories, e.g. `LoadImageFromFile` from MMCV.
+2. The transform wrapper for multiple views of an image. It is defined in [wrappers.py](../../../mmselfsup/datasets/transforms/wrappers.py).
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/transforms.md]: github link

File: docs/en/advanced_guides/transforms.md [hunk 4]
```diff
@@ -1 +1,59 @@
...
+3. The transform to pack data into a format compatible with the inputs of the algorithm. It is defined in [formatting.py](../../../mmselfsup/datasets/transforms/formatting.py).
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/transforms.md]: github link

File: docs/en/advanced_guides/transforms.md [hunk 5]
```diff
@@ -1 +1,59 @@
...
+              dict(type='Resize', scale=224),
+              dict(type='RandomSolarize')],
+          ])
+]
+```
+This means that there are two pipelines, which contain 2 views and 6 views, respectively.
+More examples can be found in [imagenet_mocov1.py](../../../configs/selfsup/_base_/datasets/imagenet_mocov1.py), [imagenet_mocov2.py](../../../configs/selfsup/_base_/datasets/imagenet_mocov2.py) and [imagenet_swav_mcrop-2-6.py](../../../configs/selfsup/_base_/datasets/imagenet_swav_mcrop-2-6.py) etc.  
+
+## Introduction of `PackSelfSupInputs`
+We build a class named `PackSelfSupInputs` to pack data into a format compatible with the inputs of the algorithm. This transform
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/transforms.md]:
```suggestion
We build a class named [`PackSelfSupInputs`](mmselfsup.datasets.transforms.PackSelfSupInputs) to pack data into a format compatible with the inputs of the algorithm. This transform
```

File: docs/en/advanced_guides/transforms.md [hunk 6]
```diff
@@ -1 +1,59 @@
...
+            [   
+              dict(type='Resize', scale=224)],
+            [
+              dict(type='Resize', scale=224),
+              dict(type='RandomSolarize')],
+          ])
+]
+```
+This means that there are two pipelines, which contain 2 views and 6 views, respectively.
+More examples can be found in [imagenet_mocov1.py](../../../configs/selfsup/_base_/datasets/imagenet_mocov1.py), [imagenet_mocov2.py](../../../configs/selfsup/_base_/datasets/imagenet_mocov2.py) and [imagenet_swav_mcrop-2-6.py](../../../configs/selfsup/_base_/datasets/imagenet_swav_mcrop-2-6.py) etc.
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/transforms.md]: github link

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: docs/en/advanced_guides/transforms.md (hunk 1 above)
Comment (wangbo-zhao) [reply docs/en/advanced_guides/transforms.md]: done

File: docs/en/advanced_guides/transforms.md (hunk 2 above)
Comment (wangbo-zhao) [reply docs/en/advanced_guides/transforms.md]: done

File: docs/en/advanced_guides/transforms.md (hunk 3 above)
Comment (wangbo-zhao) [reply docs/en/advanced_guides/transforms.md]: done

File: docs/en/advanced_guides/transforms.md (hunk 4 above)
Comment (wangbo-zhao) [reply docs/en/advanced_guides/transforms.md]: done

File: docs/en/advanced_guides/transforms.md (hunk 5 above)
Comment (wangbo-zhao) [reply docs/en/advanced_guides/transforms.md]: done

File: docs/en/advanced_guides/transforms.md [hunk 7]
```diff
@@ -1 +1,83 @@
...
+| [`RandomResizedCropAndInterpolationWithTwoPic`](mmselfsup.datasets.RandomResizedCropAndInterpolationWithTwoPic) | Crop the given PIL Image to random size and aspect ratio with random interpolation |
+|                              [`RandomSolarize`](mmselfsup.datasets.RandomSolarize)                              |                     Solarization augmentation refers to `BYOL`                     |
+|                          [`RotationWithLabels`](mmselfsup.datasets.RotationWithLabels)                          |                                Rotation prediction                                 |
+|                             [`MultiView`](mmselfsup.datasets.transforms.MultiView)                              |               A wrapper for algorithms with multi-view image inputs                |
+|                           [`PackSelfSupInputs`](mmselfsup.datasets.PackSelfSupInputs)                           |         Pack data into a format compatible with the inputs of an algorithm         |
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/transforms.md:36]: `RandomPatchWithLabels` and `RandomRotation` have not been listed

Comment (wangbo-zhao) [reply docs/en/advanced_guides/transforms.md:36]: done

**Review by fangyixiao18 — APPROVED** [review]

**Pull request merged**